.editorconfig

```diff
@@ -1052,6 +1061,9 @@ dotnet_diagnostic.RCS1159.severity = warning
 # Unused type parameter.
 dotnet_diagnostic.RCS1164.severity = warning
 
+# Use read-only auto-implemented property.
+dotnet_diagnostic.RCS1170.severity = warning
```

Comment:
it looks like this rule was implemented incorrectly. Private setters were made public. 

I'm not sure how this rule exactly works as it affected so little code, but this could make it harder to work with magic. In trait infos we use auto-implemented properties for non-yaml fields

Reply:
Ah mb I misunderstood, but it did indeed remove the privateness from RadarWidget

Reply:
The rule wants to change `public string SoundUp { get; private set; }` to `public string SoundUp { get; }`. However, this breaks our use of reflection in `FieldLoader`/`ObjectCreator` as it sets `SoundUp` from YAML.

I manually changed it to a field, like other things that need to be loaded from YAML, as it was the most elegant workaround I could see. This seemed like a worthwhile tradeoff so we can still enable an otherwise useful rule.

Reply:
Looking at git blame yeah this is supposed to be used by reflection. I thought it was purposefully added to hide it from the reflection. https://github.com/OpenRA/OpenRA/commit/85b9cf0a698a983233887993b3abcab68ae9541b https://github.com/OpenRA/OpenRA/commit/7eb64ea6fc04b692f38e2ce89c8d121ce9ab4b44

Reply:
actually should have made these readonly to match other yaml defined values